src/Anywhere.ArcGIS/Common/IGeometry.cs

```diff
@@ -439,9 +439,37 @@ public List<Point> Points
         {
             get
             {
-                return this.Select(point => point != null ? new Point { X = point.First(), Y = point.Last() } : null)
-                    .Where(p => p != null)
+
+                return this.Select(point =>
+                {
+                    if (point != null)
+                    {
+                        //If point has Z coordinate. Use it.
+                        if (point.Length == 3)
+                        {
+                            return new Point
+                            {
+                                X = point[0],
+                                Y = point[1],
+                                Z = point[2]
+                            };
+                        }
+
+                        else
+                        {
+                            //Why use "First()" and "Last()" instead of [0] and [1]?
```

Comment:
you can change to 0 and 1, just add a length check for this too

Reply:
Added it and fixed Clone() to work with Z and M values. Not sure what to do with points that have M and but no Z value. Could add a new property like 
```
public List<Point> PointsWithM
        {
            get
            {
                return this.Select(point =>
                {
                    Point result = null;
                    if (point != null && point.Length >= 2)
                    {
                        result = new Point { X = point[0], Y = point[1] };
                        //Assumption: if point has length 3 it has a M coordinate.
                        if (point.Length == 3)
                        {
                            result.M = point[2];
                        }
                        //Assumption: if point has length 4 it has a Z and M coordinate. 
                        if (point.Length == 4)
                        {
                            result.Z = point[2];
                            result.M = point[3];
                        }
                    }
                    return result;
                }).Where(p => p != null).ToList();
            }
        }
```